plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py

```diff
@@ -0,0 +1,381 @@
+
+        post_body = {
+            "request_data": {
+                "search_from": search_from,
+                "search_to": search_to,
+                "sort": {"field": time_sort_field, "keyword": "asc"},
+            }
+        }
+
+        url = urllib.parse.urljoin("https://api-rapid7.xdr.us.paloaltonetworks.com", endpoint)
+
+        # Local debugging
+        self.logger.info(f"State in UpDate: {state = }")
+        self.logger.info(f"{search_from = }")
+        self.logger.info(f"{search_to = }")
+        self.logger.info(f"{headers = }")
+        self.logger.info(f"{post_body = }")
+        self.logger.info(f"{url = }")
+
+        try:
+            request = requests.Request(method="post", url=url, headers=headers, json=post_body)
+            response = make_request(
+                request, exception_custom_configs=custom_config, timeout=120, allowed_status_codes=[]
+            )
+
+            response = response.json()
+
+            total_count = response.get("reply", {}).get("total_count", -1)
+            result_count = response.get("reply", {}).get("result_count", -1)
+            results = response.get("reply", {}).get(response_alerts_field, [])
+
+            is_paginating = ALERT_LIMIT < total_count
+
+            if is_paginating:
+                has_more_pages = True
+                state[LAST_SEARCH_FROM] = search_from
+                state[LAST_SEARCH_TO] = search_to
+                # state = {search_from: 0, search_to: 100}
+                # next run it should then be
+                # state = {search_from: 101, search_to: 200} or state = {search_from: 101, search_to: (total - search_from) (109 or whatever it equals)}
+                # and so on
+
+                self.logger.info(
+                    f"Found total alerts={total_count}, limit={ALERT_LIMIT}, is_paginating={is_paginating}"
+                )
+                self.logger.info(
+                    f"Paginating alerts: Saving state with existing filters: "
+                    f"from_time={search_from}"
+                    f"to_time={search_to}"
+                )
+
+            else:
+                has_more_pages = False
+                state = self._drop_pagination_state(state)
+
+            return results, state, has_more_pages
+
+        except PluginException as error:
+            self.logger.error(
+                f"A PluginException has occurred. Status code 500 returned. Error: {error.cause}. "
+                f"Existing state: {state}"
+            )
+            return [], state, False, 500, PluginException(cause=error.cause, data=error)
+
+        except Exception as error:
+            print(f"{error = }")
+            return [], state, False, 500, PluginException(preset=PluginException.Preset.UNKNOWN, data=error)
+
+    ###########################
+    # Deduping
+    ###########################
+    def _dedupe_and_get_highest_time(self, alerts: list, start_time: str, state: dict) -> Tuple[list, list, str]:
```

Comment:
We don't seem to call this method yet, will we have logic to get the old_hashes?

Reply:
I have made changes to use this method. Will look into whether the `old_hashes` are being used.

Reply:
Update: It is now being used 😄